extensions/wikia/PortableInfobox/tests/nodes/NodeDataTest.php

```diff
@@ -39,6 +39,47 @@ public function sourceDataProvider() {
 	}
 
 	/**
+	 * @covers       \Wikia\PortableInfobox\Parser\Nodes\Node::getSourceLabel
+	 * @dataProvider sourceLabelDataProvider
+	 *
+	 * @param $markup
+	 * @param $params
+	 * @param $expected
+	 */
+	public function testSourceLabel( $markup, $params, $expected ) {
+		$node = \Wikia\PortableInfobox\Parser\Nodes\NodeFactory::newFromXML( $markup, $params );
+
+		$this->assertEquals( $expected, $node->getSourceLabel() );
+	}
+
+	public function sourceLabelDataProvider() {
+		return [
+			[ '<data source="test"></data>', [ ], ['test' => ''] ],
+			[ '<image source="test"/>', [ ], ['test' => ''] ],
+			[ '<title source="test"/>', [ ], ['test' => ''] ],
+			[ '<data source="test"><default>{{{test}}}</default></data>',
+				[ ], ['test' => ''] ],
+			[ '<data source="test"><label source="test">{{{test}}}</label><default>{{{test}}}</default></data>',
+				[ ], ['test' => '{{{test}}}'] ],
+			[ '<data source="test"><label>testLabel</label></data>', [ ], ['test' => 'testLabel'] ],
+			[ '<data></data>', [ ], [ ] ],
+			[ '<group><data source="test"><label>labelInsideGroup</label></data></group>', [], ['test' =>'labelInsideGroup'] ],
+			[ '<group>' .
+				'<data source="test"><label>labelInsideGroup</label></data>' .
+				'<data source="test2"><label>labelInsideGroup2</label></data>' .
+				'</group>',
+				[], ['test' =>'labelInsideGroup', 'test2' =>'labelInsideGroup2'] ],
+			[ '<group>' .
+				'<title source="title"/>' .
+				'<image source="image"/>' .
+				'<data source="test"><label>labelInsideGroup</label></data>' .
+				'<data source="test2"><label>labelInsideGroup2</label></data>' .
+				'</group>',
+				[], ['test' =>'labelInsideGroup', 'test2' =>'labelInsideGroup2', 'title' => '', 'image' => ''] ]
+		];
```

Comment:
Please add more tests, especially for edge cases like use of ``<format>`` or ``<caption>`` tags.